openshift-client/src/main/java/io/fabric8/openshift/client/DefaultOpenShiftClient.java

```diff
@@ -147,17 +146,22 @@ public DefaultOpenShiftClient(String masterUrl) throws KubernetesClientException
   protected DefaultOpenShiftClient(OkHttpClient httpClient, OpenShiftConfig config) throws KubernetesClientException {
     super(httpClient, config);
     try {
-      addOpenShiftAuthInterceptor();
-      this.delegate = new DefaultKubernetesClient(httpClient, config);
+      this.delegate = new DefaultKubernetesClient(clientWithOpenShiftOAuhtInterceptor(httpClient), config);
       this.openShiftUrl = new URL(config.getOpenShiftUrl());
     } catch (MalformedURLException e) {
       throw new KubernetesClientException("Could not create client", e);
     }
   }
 
-  private void addOpenShiftAuthInterceptor() {
-    httpClient = httpClient.newBuilder().authenticator(Authenticator.NONE).build();
-    httpClient = httpClient.newBuilder().addInterceptor(new OpenShiftOAuthInterceptor(httpClient,  OpenShiftConfig.wrap(getConfiguration()))).build();
```

Comment:
Can you do `builder.interceptors().add(0, new OpenShiftOAuthInterceptor(httpClient,  OpenShiftConfig.wrap(getConfiguration())));` without the clear?